python_modules/dagster/dagster/core/object_store.py

```diff
@@ -95,6 +102,26 @@ def rm_object(self, context, paths):  # pylint: disable=unused-argument
         os.unlink(target_path)
         return
 
+    def copy_object_from_prev_run(
+        self, context, previous_run_id, paths
+    ):  # pylint: disable=unused-argument
+        prev_run_files_dir = get_run_files_directory(previous_run_id)
+        check.invariant(os.path.isdir(prev_run_files_dir))
+
+        copy_from_path = os.path.join(prev_run_files_dir, *paths)
+        copy_to_path = get_valid_target_path(self.root, paths)
+
+        check.invariant(
+            not os.path.exists(copy_to_path), 'Path already exists {}'.format(copy_to_path)
+        )
+
+        if os.path.isfile(copy_from_path):
```

Comment:
I think we have a little confusion throughout the object store implementations whether we expect to find directories at these paths or not

Reply:
they can be either. in a pickle case it could be a file but in the parquet case it could be an entire directory structure